File: myExpenses/src/main/java/org/totschnig/myexpenses/viewmodel/data/PaymentMethod.java
package org.totschnig.myexpenses.viewmodel.data;

import android.database.Cursor;

import com.gabrielittner.auto.value.cursor.ColumnName;
import com.google.auto.value.AutoValue;

import static org.totschnig.myexpenses.provider.DatabaseConstants.KEY_IS_NUMBERED;
import static org.totschnig.myexpenses.provider.DatabaseConstants.KEY_LABEL;
import static org.totschnig.myexpenses.provider.DatabaseConstants.KEY_ROWID;

//TODO convert to kotlin data class
@AutoValue
public abstract class PaymentMethod {
  @ColumnName(KEY_ROWID)
  public abstract long id();
  @ColumnName(KEY_LABEL)
  public abstract String label();
  @ColumnName(KEY_IS_NUMBERED)
  public abstract boolean isNumbered();

  public static PaymentMethod create(Cursor cursor) {
    return AutoValue_PaymentMethod.createFromCursor(cursor);
  }

  @Override
  public String toString() {
    return label();
  }
}

File: myExpenses/src/main/java/org/totschnig/myexpenses/viewmodel/data/PaymentMethod.kt
package org.totschnig.myexpenses.viewmodel.data

import android.database.Cursor
import org.totschnig.myexpenses.provider.KEY_IS_NUMBERED
import org.totschnig.myexpenses.provider.KEY_LABEL
import org.totschnig.myexpenses.provider.KEY_ROWID
import org.totschnig.myexpenses.provider.getBoolean
import org.totschnig.myexpenses.provider.getLong
import org.totschnig.myexpenses.provider.getString

data class PaymentMethod(
    val id: Long,
    val label: String,
    val isNumbered: Boolean
) {
    override fun toString(): String {
        return label
    }

    companion object {
        fun from(cursor: Cursor): PaymentMethod {
            return PaymentMethod(
                id = cursor.getLong(KEY_ROWID),
                label = cursor.getString(KEY_LABEL),
                isNumbered = cursor.getBoolean(KEY_IS_NUMBERED)
            )
        }
    }
}
